core/src/main/scala/caliban/wrappers/FieldMetrics.scala

```diff
@@ -34,68 +57,86 @@ object FieldMetrics {
   ): Wrapper.EffectfulWrapper[Any] =
     Wrapper.EffectfulWrapper(
       for {
-        ref <- Ref.make(List.empty[Timing])
-      } yield fieldDuration(totalLabel, durationLabel, ref, buckets, extraLabels)
+        timings  <- Ref.make(List.empty[Timing])
+        failures <- Ref.make(List.empty[String])
+        metrics   = new Metrics(totalLabel, durationLabel, buckets, extraLabels)
+      } yield overallWrapper(timings, failures, metrics) |+| fieldWrapper(timings, failures)
     )
 
-  private def fieldDuration(
-    totalLabel: String,
-    durationLabel: String,
-    ref: Ref[List[Timing]],
-    buckets: Histogram.Boundaries,
-    extraLabels: Set[MetricLabel]
-  ): Wrapper.FieldWrapper[Any] =
+  private def overallWrapper(
+    timings: Ref[List[Timing]],
+    failures: Ref[List[String]],
+    metrics: Metrics
+  ): OverallWrapper[Any] =
+    new OverallWrapper[Any] {
+      def wrap[R1](
+        process: GraphQLRequest => ZIO[R1, Nothing, GraphQLResponse[CalibanError]]
+      ): GraphQLRequest => ZIO[R1, Nothing, GraphQLResponse[CalibanError]] =
+        (request: GraphQLRequest) =>
+          process(request) <*
+            (for {
+              _          <- failures.get.flatMap(metrics.recordFailures)
+              timings    <- timings.get
+              nodeOffsets = resolveNodeOffsets(timings)
+              _          <- metrics.recordSuccesses(nodeOffsets, timings)
+            } yield ()).forkDaemon
```

Comment:
I'm not sure how I feel about forking this effect, but it's by far the most performant implementation. With forking, it becomes faster than the `ApolloTracing` wrapper

Reply:
Makes sense to not make the execution slower because we're updating the metrics. And since this wrapper does't wrap pure fields, we don't create too many useless fibers.

Reply:
I actually think being able to separate the measurements from the final calculation/reporting of metrics is the most beautiful and clever part of this change, so I definitely think we should keep the `forkDaemon`. :)

Also, a single `forkDaemon` call is likely to pale in comparison to all other `fork` calls during query execution.